File: board-back/src/main/java/com/example/boardback/service/UserService.java
package com.example.boardback.service;

import org.springframework.http.ResponseEntity;

import com.example.boardback.dto.response.user.GetSignInUserResponseDto;
import com.example.boardback.dto.response.user.GetUserResponseDto;

public interface UserService {

  ResponseEntity<? super GetUserResponseDto> getUser(String email);
  ResponseEntity<? super GetSignInUserResponseDto> getSignInUser(String email);

}

File: board-back-nest/src/modules/user/user.service.ts
import { Injectable } from '@nestjs/common';
import { GetSignInUserResponseDto, GetUserResponseDto } from './dto/response';
import { UserRepository } from 'modules/data-access/repository';

@Injectable()
export class UserService {

  constructor(
    private readonly userRepository : UserRepository
  ){}

  async getUser(email:string):Promise<GetUserResponseDto> {

    const userEntity = await this.userRepository.findByEmail(email);
    if(!userEntity) GetUserResponseDto.noExistUser();

    return GetUserResponseDto.success(userEntity);
  }

  async getSignInUser(email:string):Promise<GetSignInUserResponseDto> {

    const userEntity = await this.userRepository.findByEmail(email);
    if(!userEntity) GetSignInUserResponseDto.noExistUser();
    
    return GetSignInUserResponseDto.success(userEntity);
  }

}
